src/audio/module_adapter/module/cadence.c

```diff
@@ -121,8 +121,20 @@ static int cadence_codec_resolve_api(struct processing_module *mod)
 	struct cadence_codec_data *cd = module_get_private_data(mod);
 	uint32_t api_id = CODEC_GET_API_ID(DEFAULT_CODEC_ID);
 	uint32_t n_apis = ARRAY_SIZE(cadence_api_table);
+#if CONFIG_IPC_MAJOR_4
```

Comment:
Can you explain a little bit the context and why this change is needed?

Reply:
Current implementation does not work with ipc4. In order to enable ipc4 flow api id has to be retrieved from initial configuration.

Reply:
I don't like this. You are taking the easy route now, sprinkling ifdefs all over the place :) . This can hurt us on the long term. 

Can you find a solution and make it work in the same way it works for IPC3? Otherwise, give a thorough explanation in the commit message why the current solution doesn't work for IPC4 and why we need to change it.

:)

Reply:
I am trying to figure out how to obtain any better looking solution at the moment. Initial configuration payload has different content and is serviced in different way for ipc3 and ipc4. I could not find better place than init function to allocate memory for all necessary configuration. There are more protocol principal differences which makes it more difficult to substitute 1:1 ipc3/ipc4 . For example I am not sure if it is right direction  to resolve codec id from extended stream parameters which are not available in ipc4.
By now I could propose rewriting two separate functions for ipc3 and ipc4 which could make it more readable. Would it look better for you ? :-)

I see that similar approach is used in EQ_FIR module.